packages/toolshed/routes/ai/llm/generateText.ts

```diff
@@ -17,6 +19,65 @@ const JSON_SYSTEM_PROMPTS = {
     "You must respond with pure, correct JSON only - no text descriptions, no ```json code blocks, and no formatting outside of valid JSON. Your entire response should be a valid JSON object that can be parsed directly by JSON.parse() with no additional processing.",
 };
 
+/**
+ * Convert JSONSchema to Zod schema for AI SDK tool definitions
+ * This is a basic implementation that covers common cases
+ */
+function convertJSONSchemaToZod(schema: JSONSchema): z.ZodSchema {
```

Comment:
I am suspicious this is a pointless / avoidable step. Perhaps the Vercel AI SDK already speaks JSONSchema? Will revisit before landing.